src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -11,13 +12,15 @@
 namespace ProtoCore.Namespace
 {
 
-    public class ElementRewriter
+    public class ElementRewriter : AstReplacer
     {
         private readonly ClassTable classTable;
-        
-        internal ElementRewriter(ClassTable classTable)
+        private readonly ElementResolver elementResolver;
+
+        internal ElementRewriter(ClassTable classTable, ElementResolver resolver)
         {
             this.classTable = classTable;
+            this.elementResolver = resolver;
```

Comment:
Is the elemenResolver mainly for caching for faster lookups? Because i would think the actual class names are already in the class tables